source/postprocess/geoid.cc

```diff
@@ -739,6 +739,125 @@ namespace aspect
+      if (also_output_gravity_anomaly == true)
+        {
+          // have a stream into which we write the gravity anomaly data. the text stream is then
+          // later sent to processor 0
+          std::ostringstream output_gravity_anomaly;
+          // Compute the grid gravity anomaly based on spherical harmonics
+          std::vector<double> gravity_anomaly;
+          for (unsigned int i=0; i<surface_cell_spherical_coordinates.size(); ++i)
+            {
+              int ind = 0;
+              double gravity_value = 0;
+              for (signed int ideg =  min_degree; ideg < max_degree+1; ideg++)
+                {
+                  for (unsigned int iord = 0; iord < ideg+1; iord++)
+                    {
+                      // normalization after Dahlen and Tromp, 1986, Appendix B.6
+                      const std::pair<double,double> sph_harm_vals = aspect::Utilities::real_spherical_harmonic(ideg,iord,surface_cell_spherical_coordinates.at(i).first,surface_cell_spherical_coordinates.at(i).second);
+                      const double cos_component = sph_harm_vals.first; // real / cos part
+                      const double sin_component = sph_harm_vals.second; // imaginary / sine part
+
+                      // the conversion from geoid to gravity anomaly is given by gravity_anomaly = (l-1)*g/R_surface * geoid_anomaly
+                      // based on Forte (2007) equation [97]
+                      gravity_value += (geoid_coecos.at(ind)*cos_component+geoid_coesin.at(ind)*sin_component) * (ideg - 1) * surface_gravity / outer_radius * 1e5; // conversion to mgal
+                      ++ind;
+                    }
+                }
+              gravity_anomaly.push_back(gravity_value);
+            }
+
+          // Prepare the output data
+          if (output_in_lat_lon == true)
+            {
+              std::vector<std::pair<std::pair<double,double>,double> > stored_values_lon_lat;
+              double lon, lat;
+              for (unsigned int i=0; i<surface_cell_spherical_coordinates.size(); ++i)
+                {
+                  // Transfer the spherical coordinates to geographical coordinates
+                  lat = 90. - surface_cell_spherical_coordinates.at(i).first*(180./numbers::PI);
+                  lon = (surface_cell_spherical_coordinates.at(i).second <= numbers::PI
+                         ?
+                         surface_cell_spherical_coordinates.at(i).second*(180./numbers::PI)
+                         :
+                         surface_cell_spherical_coordinates.at(i).second*(180./numbers::PI) - 360.);
+                  stored_values_lon_lat.emplace_back(std::make_pair(lon,lat),gravity_anomaly.at(i));
+                }
+              // Write the solution to the stream output
+              for (unsigned int i=0; i<stored_values_lon_lat.size(); ++i)
+                {
+                  output_gravity_anomaly << stored_values_lon_lat.at(i).first.first
```

Comment:
Could you not just compute the geographical coordinate for every point inside this loop and immediately output it point-by-point? Then you would not need to create a new container holding all the converted coordinates and gravity values (you could remove the whole for-loop above.